### PR TITLE
I3tweaks

### PR DESCRIPTION
**Pull request by sicelo**
This code changes provides a scripted workaround for the fact that the method used by i3status to get battery capacity returns erroneous values for N900/bq27200. Details on the problem are at https://github.com/postmarketOS/pmbootstrap/pull/1450#issuecomment-386832258 and further information on the workaround, https://github.com/postmarketOS/pmbootstrap/pull/1450#issuecomment-386876774

---
[x] Merge on GitHub (see <https://postmarketos.org/merge>)

### CODE REVIEW

**Comment by sicelo** [comment]
Forgot to mention - there are just a few additions for using workspaces, and removing seconds display on i3status' time display

File: aports/device/device-nokia-n900/i3wm/i3status.conf [hunk 1]
```diff
@@ -12,11 +12,12 @@ general {
         color_good = "#009900"
         color_bad = "#990000"
         color_degraded = "#999900"
+	output_format = "i3bar"
```
Comment (ollieparanoid) [inline aports/device/device-nokia-n900/i3wm/i3status.conf]: Indentation is different than the other lines (tab instead of 8 spaces).

File: aports/device/device-nokia-n900/i3wm/i3status.conf [hunk 2]
```diff
@@ -12,11 +12,12 @@ general {
...
 }
 
 order += "disk /"
 order += "wireless _first_"
-order += "battery all"
+#order += "battery 0"
```
Comment (ollieparanoid) [inline aports/device/device-nokia-n900/i3wm/i3status.conf]: Please delete obsolete code instead of commenting it out.

File: aports/device/device-nokia-n900/i3wm/i3status.conf [hunk 3]
```diff
@@ -25,12 +26,16 @@ wireless _first_ {
...
-battery all {
-        format = "%status %percentage %remaining"
-}
+#battery 0 {
+#        format = "%status %percentage"
+#	path   = "/sys/class/power_supply/bq27200-0/uevent"
+#	status_unk = BAT
+#	status_chr = CHG
+#	last_full_capacity = true
+#}
```
Comment (ollieparanoid) [inline aports/device/device-nokia-n900/i3wm/i3status.conf]: more obsolete code

File: aports/device/device-nokia-n900/i3wm/i3wm.conf [hunk 4]
```diff
@@ -3,7 +3,12 @@ font pango:monospace 8
 new_window 1pixel
 
 mode "workspace" {
-        bindsym Left workspace prev
+	bindsym a workspace 1
+	bindsym s workspace 2
+	bindsym d workspace 3
+	bindsym f workspace 4
+	bindsym g workspace 5
+	bindsym Left workspace prev
```
Comment (ollieparanoid) [inline aports/device/device-nokia-n900/i3wm/i3wm.conf]: indentation

File: aports/device/device-nokia-n900/i3wm/i3wm.conf [hunk 5]
```diff
@@ -44,7 +49,7 @@ bindsym XF86ScreenSaver exec --no-startup-id lock.sh
 # finds out, if available)
 bar {
         position top
-        status_command i3status --conf ~/.config/i3/i3status.conf
+	status_command exec ~/.config/i3/bq27200_status.sh
```
Comment (ollieparanoid) [inline aports/device/device-nokia-n900/i3wm/i3wm.conf]: indentation

File: aports/device/device-nokia-n900/i3wm/bq27200_status.sh [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+# general {
+#   output_format = i3bar
+# }
+#
+# i3 config looks like this:
+# bar {
+#   status_command exec ~/.config/i3/bq27200_status.sh
+# }
+#
+# and bq27200_status.sh should be saved in ~/.config/i3/ and made executable
```
Comment (ollieparanoid) [inline aports/device/device-nokia-n900/i3wm/bq27200_status.sh]: Do we need the usage instructions here? After all we install it to the right place and configure i3 to use it already.

File: aports/device/device-nokia-n900/i3wm/bq27200_status.sh [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+get_battery(){
+	read capacity < "${bat_path}capacity"
+	read chg_full < "${bat_path}charge_full"
+	read voltage < "${bat_path}voltage_now"
+	read disch_rate < "${bat_path}current_now"
+	notification="$capacity% of $(( (chg_full / 1000)))mAh ( $(( (voltage / 1000)))mV at $(( ( disch_rate / 1000 )))mA )"
+}
+
+/usr/bin/i3status -c /home/user/.config/i3/i3status.conf | ( read line && echo "$line" && read line && echo "$line" && read line && echo "$line" && get_battery && while :
```
Comment (ollieparanoid) [inline aports/device/device-nokia-n900/i3wm/bq27200_status.sh]: How about documenting what this line does?

File: aports/device/device-nokia-n900/i3wm/bq27200_status.sh [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+do
+  read line
+  get_battery
+  echo ",[{\"full_text\":\"${notification}\" },${line#,\[}" || exit 1
```
Comment (ollieparanoid) [inline aports/device/device-nokia-n900/i3wm/bq27200_status.sh]: For consistency with `get_battery()` above, I'd also use tabs here.

File: aports/device/device-nokia-n900/i3wm/bq27200_status.sh (hunk 7 above)
Comment (sicelo) [reply aports/device/device-nokia-n900/i3wm/bq27200_status.sh]: I admit I do not fully understand it myself, but it's part of the example at https://github.com/i3/i3status/blob/master/contrib/net-speed.sh, and the code doesn't work properly without it. 

I will explain it as best as I understand it 😄

Comment (sicelo) [reply aports/device/device-nokia-n900/i3wm/bq27200_status.sh]: :champagne: i understand it completely now, finally ... anyway will sort this out tomorrow.

**Comment by sicelo** [comment]
So, when I have implemented all the changes locally, (and in sicelo:pmbootstrap), how do i bring them to this pull request? :thinking: 
or do i create a new one? that's one Github thing I don't understand yet, and the links relating to review requests don't seem to address it, unless I overlooked something

**Comment by MartijnBraam** [comment]
If you push any changes to the i3tweaks branch in your repo (even forced pushes) it will get picked up here in the PR and it will close any comment blocks that have changed code.

**Comment by sicelo** [comment]
I have no idea why those checks are failing, and the CI log doesn't give me a clue.

File: aports/device/device-nokia-n900/i3wm/i3wm.conf [hunk 9]
```diff
@@ -15,7 +20,7 @@ mode "command" {
         bindsym d exec dmenu_run; mode "default"
         bindsym q mode "default"
         bindsym k kill; mode "default"
-        bindsym b exec netsurf
+        bindsym b exec surf
```
Comment (ollieparanoid) [inline aports/device/device-nokia-n900/i3wm/i3wm.conf]: `surf` eats up way more resources than `netsurf`. Is it usable on the N900?

Comment (sicelo) [reply aports/device/device-nokia-n900/i3wm/i3wm.conf]: I will revert this ...
Actually, I thought `surf` was `netsurf`. To answer your question,  yes `surf` is very usable on N900 and the hardware keyboard makes it easy. Since `surf` has no toolbar, that saves a lot of screen real estate too :smile:

**Comment by sicelo** [comment]
Sorry guys

TLDR; Please help me fix the git mess I have created now. I think I need to revert a778981 (possibly f41faf8, 76d7c88, f166fe8, b329916, and 05f91c8 as well) which does not really have any code changes from me (explanation of how it happened is provided at the end of this message). Or, should I leave it as is? If I need to revert/rebase, please see if you can give me specifics in connection with the particular problem I have created, so that I don't make things worse than they already are.

My `git log` is as follows: 
```
*   a778981 (HEAD, my_version/i3tweaks, i3tweaks) Merge branch 'i3tweaks' of https://github.com
|\  
| * f923007 Update checksums for modified files
| * e72f254 Fix indentation and cleanup code
| * d8a106e Modify existing configs for the bq27200 fix
| * e6926fd Add wrapper for i3status for better battery status reporting on N900/bq27200
| * f41faf8 (origin/master, origin/HEAD) luna packages: bump pkgrel because of soname bump
| * 76d7c88 cross/binutils-*: update to upstream (2.30-r1)
| * f166fe8 main/linux-postmarketos-qcom: update to latest revision (#1472)
| * b329916 linux-postmarketos-qcom: Remove composite usb gadget for networking (#1462)
| * 05f91c8 main/postmarketos-artwork-wallpapers: Added the grass wallpapers (#1471)
* | 58dd8e5 Update checksums for modified files
* | 8be134f Fix indentation and cleanup code
* | c2c0c6a Modify existing configs for the bq27200 fix
* | 3802f97 Add wrapper for i3status for better battery status reporting on N900/bq27200
|/  
* c164532 (my_version/master) Add info about --method to error message (#1469)
```
How it all happened:
After commit f923007b59e6cad301729ab1e39c510c6e83c2ad, I thought I needed to pull the change in 76d7c8827bd878c1603a871ad029d5424b5208b4 onto my branch (did i really need to do it?), with the hope of later doing a last commit to address the surf vs. netsurf issue above.

So I did `git pull my_version i3tweaks`, which I now understand was wrong, as I should have used `rebase` instead. Additionally, I made some changes in connection with the surf browser issue, but forgot to `git add` the changed files, so they were not commited. The biggest mistake then followed, namely a `git push my_version i3tweaks`, which produced a778981.

**Comment by ollieparanoid** [comment]
Running this in the checked out `i3-tweaks` branch should do it:
```
$ git reset --hard e6926fdd8cbbc507f7e3e08813ef8e729686e3ab
$ git log # verify that the log is what you want to have now
$ git push --force
```

**Comment by drebrez** [comment]
@sicelo does this PR replaces the #1450 ? In this case we can close the other one

**Comment by sicelo** [comment]
@drebrez yes, if this one gets merged, I agree that #1450 should then be closed.
